### ISSUE TITLE
Testing for CommonJs env different for EventEmitter

### ISSUE THREAD

**Issue by mhkeller** [metafizzy/isotope#959]
In most places, the library checks for a CommonJs environment by checking for `exports` like [the following](https://github.com/metafizzy/isotope/blob/feff43802bb689b3ab9d05855d8f3320640367a2/dist/isotope.pkgd.js#L224-L229)

````js
if ( typeof define === 'function' && define.amd ) {
  // AMD
  define( 'eventie/eventie',eventie );
} else if ( typeof exports === 'object' ) {
  // CommonJS
  module.exports = eventie;
} else {
  // browser global
  window.eventie = eventie;
}
`````

For `EventEmitter` it checks for `module` and `module.exports` [like this](https://github.com/metafizzy/isotope/blob/feff43802bb689b3ab9d05855d8f3320640367a2/dist/isotope.pkgd.js#L693-L704)

````js
    // Expose the class either via AMD, CommonJS or the global object
    if (typeof define === 'function' && define.amd) {
        define('eventEmitter/EventEmitter',[],function () {
            return EventEmitter;
        });
    }
    else if (typeof module === 'object' && module.exports){
        module.exports = EventEmitter;
    }
    else {
        exports.EventEmitter = EventEmitter;
    }
````

I imagine this different style is just from how the [EventEmitter](https://github.com/Olical/EventEmitter) library that Isotope pulls from does it? 

The problem this is causing is I'm packaging Isotope in an [Electron](http://github.com/atom/electron) app, which reads scripts in a CommonJs environment. This can cause some issues for libs like jQuery that detect their environment and load differently. A few workarounds are discussed here https://github.com/atom/electron/issues/254 

The main problem is that Isotope loads fine as a browser global since ` typeof exports === 'object'` is `false` in the electron environment, *except* it doesn't expose `EventEmitter` to itself since the check for that, `typeof module === 'object' && module.exports`, is `true`. 

It could be that Electron isn't faithfully replicating the CommonJs environment. My solution for now is to fork the library and make sure that it loads `EventEmitter` as a browser global unless there are better suggestions or this is a candidate to be standardized.

**Comment by desandro** [comment]
Thanks for reporting this issue. @Olical perhaps EventEmitter should follow [UMD](https://github.com/umdjs/umd/blob/master/commonjsStrictGlobal.js)?

**Comment by Olical** [comment]
Huh, I thought it already did. Will confirm and swap to it if not though. Will update tonight.

**Comment by Olical** [comment]
I'm not totally sure about swapping to global UMD though. That seems pretty dangerous. Could quite easily override someone else's global property.

**Comment by desandro** [comment]
Yes, sorry. The UMD examples are hard to follow. I only mean adopting `typeof exports === 'object'`

Thanks for the response!

**Comment by mhkeller** [comment]
Yep `typeof exports === 'object'` would solve it for me. The issue was that other parts of this module weren't seeing `EventEmitter` — it didn't to be global for the app using Isotope.

I also noticed that Isotope [sometimes](https://github.com/metafizzy/isotope/blob/master/dist/isotope.pkgd.js#L1223) tests CommonJs with `==` as opposed to other places where it uses `===` if you want to use this as a moment to further standardize.

**Comment by desandro** [comment]
FWIW `==` is okay with `typeof`, as `typeof` always returns a string.

**Comment by mhkeller** [comment]
For sure, purely a style note. Feel free to disregard as overly neurotic :).

**Comment by Olical** [comment]
Sorry about not getting around to fixing this last night. I'll get it in today though. It'll be `v4.2.12`. Just adding an extra check sounds fine to me, that other UMD did have me slightly perplexed :smile:

**Comment by desandro** [comment]
no hurry. I like waiting weeks before pushing releases :stuck_out_tongue_winking_eye:

**Comment by Olical** [comment]
I've created a PR with a check for `typeof module.exports === 'object'` instead of just truthy, but I don't think that's what you wanted. Did you mean for me to check for the global object `exports`, not `module.exports`?

**Comment by desandro** [comment]
Looks like I was mistaken, the UMD recommendation is currently

``` js
typeof module === 'object' && module.exports
```

EventEmitter already has this. I recommend using the same kind of fixes suggested in atom/electron#254, rather than coercing checking for `module.exports`.

**Comment by shrewdlogarithm** [comment]
Waking this because I think I've found a related issue (tho it might be a bit specific)

I'm attempting to use a packaged (2.2.2) version of Isotope on a page which can be loaded into a browser (working fine) and into an Atom Electron (Node/Chromium packaged app) window (not working)

The Atom Electron environment is common.js and supports "require" but because the code must work in a browser I'm using <script> tags to load isotope - and I'm hitting the

"Cannot find property "prototype" of undefined" error - which relates to 'EventEmitter' - which is being undefined by the code someone listed above (around like 694 of the packaged code)

I don't know enough about the differences between common.js/node and regular browsers to fix this myself but I thought I'd flag it as it's likely one of those 'use cases' of the code which fails?

**Comment by mhkeller** [comment]
My quick fix for this was 
https://github.com/newslynx/isotope/commit/021f453f8846a2c0ce5d359bc0698c27459617fd 
https://github.com/newslynx/newslynx-app/commit/6f585035b978512dc6f2a0f3e5dddbf04415cb88

But a more robust solution would be good.
